Documentation/README.md

```diff
@@ -0,0 +1,22 @@
+# Documentation
+
+## User Documentation
+
+The following documentation documents are primarily intended for users of SourceKit-LSP.
+
+- [Configuration File](Configuration%20File.md)
+- [Diagnose Bundle](Diagnose%20Bundle.md)
+- [Editor Integration](Editor%20Integration.md)
+- [Enable Experimental Background Indexing](Enable%20Experimental%20Background%20Indexing.md)
+- [Environment Variables](Environment%20Variables.md)
+
+## Developer Documentation
+
+The following documentation documents are primarily intended for developers of SourceKit-LSP.
```

Comment:
Should we link to these from CONTRIBUTING.md instead?

Reply:
I split the documentation into the `Documentation` folder (documentation for users of SourceKit-LSP) and `Contributor Documentation` (for contributors) and link to `Contributor Documentation` from CONTRIBUTING.md.